privacyidea/lib/eventhandler/tokenhandler.py

```diff
@@ -405,6 +413,8 @@ def do(self, action, options=None):
                         if not init_param['email']:
                             log.warning("Enrolling EMail token. But the user {0!s}"
                                         "has no email address!".format(user))
+                    if handler_options.get("smtp_identifier"):
+                        init_param["email.identifier"] = handler_options.get("smtp_identifier")
```

Comment:
Please add a test for this code line.

You can add the test here:
https://github.com/privacyidea/privacyidea/blob/master/tests/test_lib_events.py#L1802
where an event handler is created. You can add one, that also has the "smtp_identifier" as an option.